flink-python/pyflink/datastream/stream_execution_environment.py

```diff
@@ -711,6 +713,17 @@ def from_collection(self, collection: List[Any],
         :param type_info: The TypeInformation for the produced data stream
         :return: the data stream representing the given collection.
         """
+        if type_info is None:
+            if any(isinstance(element, (dict, tuple, list)) for element in collection) or \
+                    any(hasattr(element, "__dict__") for element in collection):
+                data_type = _infer_schema_from_data(collection)
+            else:
+                data_type = _infer_type(collection)
+            converter = _create_converter(data_type)
```

Comment:
Maybe we can design a covert util for DataStream collection elements based on user provided typeInfo. Introducing   DataTypes for Table/SQL API in DataStream API is also a bit confusing.

Reply:
@shuiqiangchen Yes, I extend the WrapperTypeInfo to support to convert the wrapper object.